packages/naming.cell.match/cell-match.js

```diff
@@ -0,0 +1,102 @@
+'use strict';
+
+const assert = require('assert');
+
+const BemCell = require('@bem/sdk.cell');
+const bemNamingParse = require('@bem/sdk.naming.entity.parse');
+const pathPatternParser = require('@bem/sdk.naming.cell.pattern-parser');
+
+const ALPHANUM_RE = '[a-z][\\w\\-]*';
+const resc = s => String(s).replace(/[\\^$*+?.()|[\]{}]/g, '\\$&');
+
+const buildPathParseMethod = ({ fs: { pattern, delims: dd = {}, scheme = 'nested' }, delims, wordPattern: wp = ALPHANUM_RE }) => {
+    const separation = pathPatternParser(pattern);
+
+    const dElem = 'elem' in dd ? dd.elem : (delims.elem || '__');
+    const dMod = 'mod' in dd ? dd.mod : (Object(delims.mod).name || (typeof delims.mod === 'string' && delims.mod) || '_');
+
+    const entityRe = ({
+        flat: [`(?:(${wp})(?:/(${ALPHANUM_RE})`, ')?)?', (entity, { dir }) => (entity.block === dir)],
+        nested: [`(?:(${wp}(?:|/${dElem}${wp})?(?:|/${dMod}${wp})?)(?:/(${ALPHANUM_RE})`, ')?)?', (entity, { dir }) => {
+            const parts = dir.split('/');
+            let i = 1;
+            return entity.block === parts[0] &&
+                (!entity.elem || (parts[i++] === dElem + entity.elem)) &&
+                (!entity.mod || (parts[i++] == dMod + entity.mod.name));
+        }]
+    })[scheme];
+
+    let deeper = [];
+    const keys = [];
+    const res = [];
+    const join = (parts, j) => {
+        for (let i = 0; i < parts.length - j; i += 1) {
+            const el = parts[i + j];
+            if (i % 2 === 0) {
+                const subParts = el.split('/');
+                res.push(subParts.map(part => resc(part)).join('(?:/'));
+                [].unshift.apply(deeper, Array.from(new Array(subParts.length - 1)).map(() => ')?'));
+            } else if (Array.isArray(el)) {
+                res.push('(?:');
+                join(el, 1);
+                res.push(')?');
+            } else if (el === 'entity') {
+                keys.push('dir', el);
+                res.push(entityRe[0]);
+                deeper.unshift(entityRe[1]);
+            } else {
+                keys.push(el);
+                res.push(`(${wp})`);
+            }
+        }
+    };
+    join(separation, 0);
+
+    const regexp = new RegExp('^' + res.concat(deeper).join('') + '(.*)$');
+    keys.push('rest');
+
+    return { regexp, keys, isValid: entityRe[2] };
+};
+
+/**
+ * Stringifier generator
+ *
+ * @param {BemNamingConvention} conv - naming, path and scheme
+ * @returns {function(BemCell): string} converts cell to file path
+ */
+module.exports = (conv = {}) => {
+    assert(conv.fs && typeof conv.fs.pattern === 'string',
```

Comment:
Why should we write it many times? [Fist time](https://github.com/bem/bem-sdk/blob/9dbb34042f7b025c8ee427f9af81fd202d889b3f/packages/naming.cell.stringify/cell-stringify.js#L41-L42). Can we put it inside `BemNaming(conv)` call?

Reply:
Fix planned in #304